Lenient YAML Parsing

cmmoran opened the pull request:
Fixes: https://github.com/ashald/EnvFile/issues/213

If EnvFile encounters a well-formed YAML configuration file that contains values other than literal strings, it will fail with a `ClassCastException`.

This change should have no effect on YAML files with string literal values but will allow EnvFile to support a wider range of configuration files without issue.

cmmoran committed:
lenient yaml parsing

mrichards01 commented on modules/core/src/main/java/net/ashald/envfile/providers/yaml/YamlEnvFileParser.java:
```diff
@@ -4,14 +4,28 @@
+            if (v != null && v.getClass().equals(String.class)) {
+                result.put(key, (String) v);
+            } else {
+                String stringValue = String.valueOf(v);
```
Just browsing the codebase and noticed your PR. If v can be null, wouldn't String.valueOf(null) throw an exception. Similarly, I think null is also meant to be a valid value in yaml files.

cmmoran committed:
chore: upgrading ide versions

cmmoran committed:
chore: upgrading ide versions

cmmoran committed:
Merge remote-tracking branch 'refs/remotes/dennisameling/upgrade-intellij-gradle-plugin-to-v2' into develop

# Conflicts:
#	build.gradle
#	src/main/resources/META-INF/plugin.xml

cmmoran committed:
refactor: remove legacy Gradle build scripts and migrate package structure to align with new maintainer

chore: update JetBrains IDE versions and simplify exception handling in build script
